### ISSUE TITLE
tests: error with golang-1.18

### ISSUE THREAD

**Issue by linuxmaniac** [asipto/secsipidx#22]
```
=== RUN   TestParseECPublicKeyFromPEM/ErrCertInvalid_with_invalid_certificate
    expect.go:34: 'x509: malformed certificate' is not 'asn1: structure error: tags don't match (16 vs {class:1 tag:9 length:110 isCompound:true}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} certificate @2'

--- FAIL: TestParseECPublicKeyFromPEM (0.00s)
--- FAIL: TestParseECPublicKeyFromPEM/ErrCertInvalid_with_invalid_certificate (0.00s)
```

**Comment by linuxmaniac** [comment]
@steve-kaufman do you have any idea how to fix this?

**Comment by miconda** [comment]
@steve-kaufman: maybe we can also get rid of depending on `github.com/gomagedon/expectate`. Can it be replaced by core Golang functions/testing extensions?

**Comment by miconda** [comment]
Tests should be fixed now, the issue was with checking error messages thrown by go tls certificate library functions. Probably the error messages should not be checked in such cases, as we do not control them and they can change with new go versions, checking the return code from secsipid functions should be enough.

The fixes are for `go 1.19.1`, considering the above, it may still fail with other go versions.

Also, probably removing expectate dependency should be done, it does not provide much benefits over all, as I could figure out.
